util/patricia_trie/src/node.rs

```diff
@@ -107,13 +108,43 @@ impl<'a> Node<'a> {
 	pub fn try_decode_hash(node_data: &[u8]) -> Option<H256> {
 		let r = Rlp::new(node_data);
 		if r.is_data() && r.size() == 32 {
-			Some(r.as_val().expect("Hash is the correct size of 32 bytes; qed"))
+			Some(
+				r.as_val()
+					.expect("Hash is the correct size of 32 bytes; qed"),
+			)
 		} else {
 			None
 		}
 	}
 }
 
+impl<'a> fmt::Debug for Node<'a> {
+	/// Recursion helper for implementation of formatting trait.
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match &self {
+			Node::Leaf(slice, value) => f.debug_struct("Node::Leaf")
+				.field("slice", &slice)
+				.field("value", &value.pretty())
```

Comment:
is it possible to avoid calling `.pretty()`? :)

Reply:
Yes, but that gives a different value: `00·01` (on pretty) vs `[0, 1]`. I am fine with the latter, but just to make sure that is okay.